### PR TITLE
(feat) SvelteKit typings without types

### PR DESCRIPTION
**Pull request by dummdidumm**
This allows you to omit the `./$types` imports and still get proper typings, by inserting them under the hood. Example:

```ts
// +page.ts
export function load(event) {}
--->
export function load(event: import('./$types').PageLoadEvent) {}
```

This works for TS files, in Svelte files, and for `svelte-check`.

Works in special SvelteKit files (`+page/layout(.server).ts/js`) for all exports (`actions`/`csr`/`ssr`/`prerender`/`trailingSlash`/`load`)

### CODE REVIEW

**Comment by dummdidumm** [comment]
FYI @jasonlyu123 I merged this so we can "beta"-test it, you're still welcome to give this a review 👍

**Comment by fehnomenal** [comment]
Nice! Could this also work with `export const load` arrow functions like the extension generates?

File: packages/typescript-plugin/src/language-service/sveltekit.ts [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+        fileExists(fileName: string) {
+            return (
+                this.files[fileName] !== undefined ||
+                originalLanguageServiceHost.fileExists(fileName)
+            );
+        }
+    }
+
+    const languageServiceHost = new ProxiedLanguageServiceHost();
+    const languageService = ts.createLanguageService(
```
[the remark this answers is not in the document]
Comment (dummdidumm) [reply packages/typescript-plugin/src/language-service/sveltekit.ts:522]: Damn this is a good find, thank you! I'll add this in a follow-up PR.

File: packages/typescript-plugin/src/language-service/sveltekit.ts [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+    logger?: Logger
+) {
+    const cachedProxiedLanguageService = cache.get(info);
+    if (cachedProxiedLanguageService) {
+        return cachedProxiedLanguageService;
+    }
+
+    const originalLanguageServiceHost = info.languageServiceHost;
+
+    class ProxiedLanguageServiceHost implements ts.LanguageServiceHost {
```
[the remark this answers is not in the document]
Comment (dummdidumm) [reply packages/typescript-plugin/src/language-service/sveltekit.ts:310]: Tried, am running into what seems caching issues (the language service host in full capacity seems to do stuff with the diagnostics differently), so will only patch `readDirectory` for now

**Pull request merged**